src/__Private/LinterCLIErrorHandlerPlain.php

```diff
@@ -0,0 +1,216 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST\__Private;
+
+use namespace Facebook\HHAST\Linters;
+use namespace HH\Lib\{C, Str, Vec};
+
+final class LinterCLIErrorHandlerPlain implements LinterCLIErrorHandler {
+
+  private bool $had_errors = false;
+
+  public function __construct(
+    private bool $supportsColors,
+    private bool $isInteractive,
```

Comment:
we generally consider bool arguments to be an antipattern as it's unclear at the callsite, especially when there's multiple bool params - perhaps a config shape, or enum values instead?